helper.go

```
@@ -107,6 +120,77 @@ func InitHelper(config *Config) (*Helper, error) {
+	if providerRootDir == "" {
+		// common case; assume intentional and do not log
+		return nil
+	}
+
+	_, err := os.Stat(filepath.Join(providerRootDir, "terraform.d", "plugins"))
+	if err != nil {
+		fmt.Printf("No terraform.d/plugins directory found: continuing. Unset TF_ACC_PROVIDER_ROOT_DIR or supply provider binaries in terraform.d/plugins/$GOOS_$GOARCH to disable this message.")
+		return nil
+	}
+
+	auxiliaryProviderDir := filepath.Join(providerRootDir, "terraform.d", "plugins", runtime.GOOS+"_"+runtime.GOARCH)
+
+	// If we can't os.Stat() terraform.d/plugins/$GOOS_$GOARCH, however,
+	// assume the omission was unintentional, and error.
+	_, err = os.Stat(auxiliaryProviderDir)
+	if err != nil {
+		return fmt.Errorf("error finding auxiliary provider dir %s: %s", auxiliaryProviderDir, err)
+	}
+
+	// now find all the providers in that dir and symlink them to the plugin dir
+	providers, err := ioutil.ReadDir(auxiliaryProviderDir)
+	if err != nil {
+		return fmt.Errorf("error reading auxiliary providers: %s", err)
+	}
+
+	zipDecompressor := new(getter.ZipDecompressor)
+
+	for _, provider := range providers {
+		providerFilename := provider.Name()
+		providerNameExt := filepath.Ext(providerFilename)
+		providerName := strings.TrimSuffix(providerFilename, providerNameExt)
+		providerPath := filepath.Join(auxiliaryProviderDir, providerName)
+		providerSymlinkPath := filepath.Join(pluginDir, providerName)
+
+		// exit early if we have already symlinked this provider
+		_, err := os.Stat(providerSymlinkPath)
+		if err == nil {
+			continue
+		}
+
+		// if filename ends in .zip, assume it is a zip and extract it
+		// otherwise assume it is a provider binary
+		if providerNameExt == ".zip" {
+			_, err = os.Stat(providerPath)
+			if err == nil {
+				// assume we have already extracted this provider zip
+			} else {
+				zipDecompressor.Decompress(providerPath, filepath.Join(auxiliaryProviderDir, providerFilename), false)
```

Are we intentionally ignoring errors here? What if the ZIP file is corrupted, password protected or something like that?